src/Microsoft.DotNet.Core.Build.Tasks/GenerateRuntimeConfigurationFiles.cs

```diff
@@ -121,10 +119,24 @@ private void AddAdditionalProbingPaths(RuntimeOptions runtimeOptions)
                 }
 
                 // DotNetHost doesn't handle additional probing paths with a trailing slash
-                runtimeOptions.AdditionalProbingPaths.Add(PathUtility.EnsureNoTrailingSlash(packageFolder.Path));
+                runtimeOptions.AdditionalProbingPaths.Add(EnsureNoTrailingSlash(packageFolder.Path));
             }
         }
 
+        private static string EnsureNoTrailingSlash(string path)
+        {
+            if (!string.IsNullOrEmpty(path))
+            {
+                char lastChar = path[path.Length - 1];
+                if (lastChar == Path.DirectorySeparatorChar || lastChar == '/')
```

Comment:
The values coming into this method are only coming from the lock file's `packageFolder`, which are written by NuGet, so we don't need to worry about MSBuild, AFAIK.

The thing is `\` is a valid character in a file/folder name on unix.  So trimming that off on unix isn't the correct thing...